src/test/crimson/seastore/test_omap_manager.cc

```diff
@@ -0,0 +1,606 @@
+    const string &key) {
+    auto ret = omap_manager->omap_rm_key(omap_root, t, key).unsafe_get0();
+    EXPECT_EQ(ret, true);
+    test_omap_mappings.erase(test_omap_mappings.find(key));
+    return ret;
+  }
+
+  list_keys_result_t list_keys(
+    omap_root_t &omap_root,
+    Transaction &t,
+    std::string &start,
+    size_t max = MAX_SIZE) {
+    auto ret = omap_manager->omap_list_keys(omap_root, t, start, max).unsafe_get0();
+    if (start == "" && max == MAX_SIZE) {
+      EXPECT_EQ(test_omap_mappings.size(), ret.keys.size());
+      for ( auto &i : ret.keys) {
+        auto it = test_omap_mappings.find(i);
+        EXPECT_NE(it, test_omap_mappings.end());
+        EXPECT_EQ(i, it->first);
+      }
+      EXPECT_EQ(ret.next, "");
+    } else {
+      size_t i =0;
+      auto it = test_omap_mappings.find(start);
+      for (; it != test_omap_mappings.end() && i < max; it++) {
+        EXPECT_EQ(ret.keys[i], it->first);
+        i++;
+      }
+      if (it == test_omap_mappings.end()) {
+        EXPECT_EQ(ret.next, "");
+      } else {
+        EXPECT_EQ(ret.keys.size(), max);
+        EXPECT_EQ(ret.next, it->first);
+      }
+    }
+    return ret;
+  }
+
+  list_kvs_result_t list(
+    omap_root_t &omap_root,
+    Transaction &t,
+    std::string &start,
+    size_t max = MAX_SIZE) {
+    auto ret = omap_manager->omap_list(omap_root, t, start, max).unsafe_get0();
+    if (start == "" && max == MAX_SIZE) {
+      EXPECT_EQ(test_omap_mappings.size(), ret.kvs.size());
+      for ( auto &i : ret.kvs) {
+        auto it = test_omap_mappings.find(i.first);
+        EXPECT_NE(it, test_omap_mappings.end());
+        EXPECT_EQ(i.second, it->second);
+      }
+      EXPECT_EQ(ret.next, "");
+    } else {
+      size_t i = 0;
+      auto it = test_omap_mappings.find(start);
+      for (; it != test_omap_mappings.end() && i < max; it++) {
+        EXPECT_EQ(ret.kvs[i].first, it->first);
+        i++;
+      }
+      if (it == test_omap_mappings.end()) {
+        EXPECT_EQ(ret.next, "");
+      } else {
+        EXPECT_EQ(ret.kvs.size(), max);
+        EXPECT_EQ(ret.next, it->first);
+      }
+    }
+
+    return ret;
+  }
+
+  void clear(
+    omap_root_t &omap_root,
+    Transaction &t) {
+    omap_manager->omap_clear(omap_root, t).unsafe_get0();
+    EXPECT_EQ(omap_root.omap_root_laddr, L_ADDR_NULL);
+  }
+
+  void check_mappings(omap_root_t &omap_root, Transaction &t) {
+    for (const auto &i: test_omap_mappings){
+      auto ret = get_value(omap_root, t, i.first);
+      EXPECT_EQ(i.first, ret.first);
+      EXPECT_EQ(i.second, ret.second);
+    }
+  }
+
+  void check_mappings(omap_root_t &omap_root) {
+    auto t = tm->create_transaction();
+    check_mappings(omap_root, *t);
+  }
+
+  void replay() {
+    logger().debug("{}: begin", __func__);
+    tm->close().unsafe_get();
+    destroy();
+    static_cast<segment_manager::EphemeralSegmentManager*>(&*segment_manager)->remount();
+    init();
+    tm->mount().unsafe_get();
```

Comment:
nit, reuse `TMTestState::restart()`.

Reply:
where is the restart() function, i didn't found it in transaction_manager_test_state.h

Reply:
see https://github.com/ceph/ceph/blob/master/src/test/crimson/seastore/transaction_manager_test_state.h#L55-L61